### ISSUE TITLE
Unable to create L2 network for user via UI

### ISSUE THREAD

**Issue by top-secrett** [apache/cloudstack#7377]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### CONFIGURATION

Advanced zone

##### OS / ENVIRONMENT

OracleLinux 8

##### SUMMARY

When I create guest L2 network with vlan as admin for another user, this L2 network creates as mine. User doesn't see it and can't use it 
![image](https://user-images.githubusercontent.com/92435079/228501825-c375e9ea-ab39-487a-93bd-ade7ef168fed.png)
![image](https://user-images.githubusercontent.com/92435079/228501864-1f8bcaa9-dd31-45bb-89ef-48415c93133b.png)

##### STEPS TO REPRODUCE

~~~
1. Open "Guest networks " - "Add network"
2. Choose "L2"
3. Choose domain and Account, fill vlan field
4. Press OK
~~~

##### EXPECTED RESULTS

~~~
L2 network created for chosen account
~~~

##### ACTUAL RESULTS

~~~
L2 network created for mine admin account
~~~

**Comment by top-secrett** [comment]
It work via CloudMonkey 
~~~
create network account=490316480 acltype=Account broadcastdomaintype=Vlan broadcasturi="vlan://681" displaytext=true domainid=6210898b-0c5c-449a-8a05-13904c2bc279 name=L2-network-VL681 networkofferingid=29380825-8a7b-4d80-8539-1b5c06c7e058 zoneid=a5a0c8c3-f4ed-48e9-b3bf-3d645c96d252 specifyVlan=true vlan=681
~~~

**Comment by kiranchavala** [comment]
Able to reproduce the issue 

The issue is not happening with Isolated and Shared network 

It's only occurs if we create a L2 network  from the UI

**Comment by harikrishna-patnala** [comment]
Fixed with #7382
